### ISSUE TITLE
no-unused-vars errors with parameters when parameter marked used later

### ISSUE THREAD

**Issue by webOS101** [eslint/eslint#10952]
**Tell us about your environment**

* **ESLint Version:** 5.6.1 
* **Node Version:** v9.4.0
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?** babel-eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
	"parser": "babel-eslint",
	"plugins": [
		"babel",
		"react"
	],
	"extends": ["plugin:react/recommended"],
	"rules": {
		"no-unused-vars": 1
	}
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

The eslint-plugin-react rule for `jsx-uses-vars` [marks the `Wrapped` parameter as used](https://github.com/yannickcr/eslint-plugin-react/blob/master/lib/rules/jsx-uses-vars.js#L45).  However, `no-unused-vars` doesn't update the status of `config`.  The default for `no-unused-vars` should not warn on parameters before the first used parameter.

```js
import React from 'react';
import foo from 'foo';

export const AccessibilityDecorator = foo((config, Wrapped) => {
	return class t extends React.Component {
		render () {
			return <Wrapped />;
		}
	};
});
```

```bash
eslint .
```

**What did you expect to happen?**

No warnings.

**What actually happened? Please include the actual, raw output from ESLint.**

```
/Users/.../.../test.js
  4:44  warning  'config' is defined but never used  no-unused-vars

✖ 1 problem (0 errors, 1 warning)
```

Linked issue: yannickcr/eslint-plugin-react#2009

**Comment by platinumazure** [comment]
Hi @webOS101, thanks for the issue.

Would you mind running ESLint with the `--print-config` option, passing in the path of the file you're trying to lint here? Curious to see if your configuration is being overridden somewhere (either in `no-unused-vars`, or `react/jsx-uses-vars`). Thanks!

**Comment by webOS101** [comment]
It may be as simple as checking `isEslintUsed` here:  https://github.com/eslint/eslint/blob/648a33a34d24de22240df99dfd68beded660a890/lib/rules/no-unused-vars.js#L472

But I didn't actually test that.

**Comment by webOS101** [comment]
I confirmed that works.  I was able to create a test case that fails and that fixes it.  I still have never figured out how to use a locally linked eslint to test with either global or local installs.  I guess I'll figure it out someday.  I'll create a PR shortly.

**Comment by platinumazure** [comment]
Thanks @webOS101 for digging further! I think I rewrote that code some time ago and this was an oversight on my part.

I've relabeled this as a bug, in accordance with your investigation results.
